src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/endtoendtests/PaymentsTest.java

```diff
@@ -0,0 +1,153 @@
+package uk.gov.hmcts.reform.bulkscan.endtoendtests;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import com.warrenstrange.googleauth.GoogleAuthenticator;
+import io.restassured.RestAssured;
+import io.restassured.response.Response;
+import org.junit.jupiter.api.Test;
+import uk.gov.hmcts.reform.bulkscan.endtoendtests.helper.Await;
+import uk.gov.hmcts.reform.bulkscan.endtoendtests.helper.Container;
+import uk.gov.hmcts.reform.bulkscan.endtoendtests.helper.StorageHelper;
+import uk.gov.hmcts.reform.bulkscan.endtoendtests.helper.ZipFileHelper;
+import uk.gov.hmcts.reform.bulkscan.endtoendtests.utils.ProcessorEnvelopeResult;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static java.lang.String.format;
+import static org.apache.http.HttpHeaders.CONTENT_TYPE;
+import static org.apache.http.HttpStatus.SC_OK;
+import static org.apache.http.entity.ContentType.APPLICATION_FORM_URLENCODED;
+import static org.apache.http.entity.ContentType.APPLICATION_JSON;
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.bulkscan.endtoendtests.utils.ProcessorEnvelopeStatusChecker.getZipFileStatus;
+
+public class PaymentsTest {
+
+    private static final String BEARER = "Bearer ";
+
+    private Config conf = ConfigFactory.load();
+
+    private ObjectMapper objectMapper = new ObjectMapper();
+
+    @Test
+    public void should_create_exception_record_with_payments_when_envelope_contains_payments() throws Exception {
+
+        var zipArchive = ZipFileHelper.createZipArchive("test-data/new-application-payments", Container.BULKSCAN);
+
+        StorageHelper.uploadZipFile(Container.BULKSCAN, zipArchive);
+
+        Await.envelopeDispatched(zipArchive.fileName);
+        Await.envelopeCompleted(zipArchive.fileName);
+
+        //get the process result again and assert
+        assertCompletedProcessorResult(zipArchive.fileName);
+        String ccdId = retrieveCcdId(zipArchive.fileName);
+
+        String accessToken = getIdamToken();
+
+        String s2sToken = getS2SToken();
+
+        Map<?, ?> caseData = getCaseData(accessToken, s2sToken, ccdId);
+
+        String awaitingPaymentDCNProcessing = (String)caseData.get("awaitingPaymentDCNProcessing");
+        String containsPayments = (String)caseData.get("containsPayments");
+        assertThat(containsPayments).isEqualTo("Yes");
+        assertThat(awaitingPaymentDCNProcessing).isEqualTo("No");
+    }
+
+    private String getIdamToken() throws com.fasterxml.jackson.core.JsonProcessingException {
```

Comment:
Can we extract the two client methods into separate class(es)?

Reply:
Done. Also introduced CcdClient